src/scripts/main.js

```diff
@@ -1,3 +1,238 @@
+      theSameCellsAdded = false;
+    }
+  }
+});
+
+function addNewCell() {
+  let startCell = '';
+
+  do {
+    startCell = fieldCells[Math.floor(Math.random() * fieldCells.length)];
+  } while (startCell.classList.length !== 1);
+
+  if (Math.random() > 0.9) {
+    startCell.classList.add('field-cell--4');
+    startCell.textContent = '4';
+  } else {
+    startCell.classList.add('field-cell--2');
+    startCell.textContent = '2';
+  }
+
+  endGameCheck();
+}
+
+function endGameCheck() {
+  const allCellsFilled = fieldCells.every(item => item.classList.length === 2);
+  let notTheSameNeighbors = true;
+
+  for (let i = 0; i < fieldRows.length; i++) {
+    for (let iInner = 0; iInner < fieldRows[i].children.length; iInner++) {
+      if (fieldRows[i].children[iInner].nextElementSibling) {
+        if (fieldRows[i]
+          .children[iInner].textContent === fieldRows[i]
+          .children[iInner].nextElementSibling.textContent) {
+          notTheSameNeighbors = false;
+        }
+      }
+
+      if (i < 3) {
+        if (fieldRows[i]
+          .children[iInner].textContent === fieldRows[i + 1]
+          .children[iInner].textContent) {
+          notTheSameNeighbors = false;
+        }
+      }
+    }
+  }
+
+  if (allCellsFilled && notTheSameNeighbors) {
+    messageLose.classList.remove('hidden');
+  }
+}
+
+function addCells(sacrifice, direction, indexRow) {
+  switch (direction) {
+    case 'ArrowUp':
+    case 'ArrowLeft':
+      sortCells(sacrifice);
+      break;
+
+    case 'ArrowDown':
+    case 'ArrowRight':
+      sacrifice.reverse();
+      sortCells(sacrifice);
+      sacrifice.reverse();
+      break;
+  }
+
+  switch (direction) {
+    case 'ArrowUp':
+    case 'ArrowDown':
+      for (let i = 0; i < sacrifice.length; i++) {
+        fieldRows[i].append(sacrifice[i]);
+      }
+      break;
+
+    case 'ArrowLeft':
+    case 'ArrowRight':
+      for (let i = 0; i < sacrifice.length; i++) {
+        fieldRows[indexRow].append(sacrifice[i]);
+      }
+      break;
+  }
+
+  function sortCells(array) {
+    array.sort((a, b) => {
+      if (a.textContent === '') {
+        return 1;
+      }
+
+      if (b.textContent === '') {
+        return -1;
+      }
+
+      if (a.textContent === b.textContent) {
+        a.classList.remove(`field-cell--${a.textContent}`);
+        a.textContent = '';
+        b.classList.remove(`field-cell--${b.textContent}`);
+        b.textContent *= 2;
+        b.classList.add(`field-cell--${b.textContent}`);
+        theSameCellsAdded = true;
+        gameScore.textContent = +gameScore.textContent + +b.textContent;
+      }
+
+      return 0;
+    });
+  }
+}
+
+function needAddNewCell(arrayRow, direction) {
+  return arrayRow.some(item => {
+    switch (direction) {
+      case 'ArrowLeft':
+        for (let i = item.children.length - 1; i > 0; i--) {
+          if (item.children[i].textContent
+            && !item.children[i - 1].textContent) {
+            return true;
+          }
```

Comment:
so return just condition. If it's true you return true
```suggestion
          return item.children[i].textContent && !item.children[i - 1].textContent;
```